### PR TITLE
Fix transactions pagination issue

### PR DESCRIPTION
**Pull request by igor-egorov**
Signed-off-by: Igor Egorov <igor@soramitsu.co.jp>

### Description of the Change

Indexes over transactions in blocks were stored as strings that resulted in lexicographically sorting of index values instead of numeric sorting.

As a result, in some cases, more transactions were shown in the response than needed.

### Benefits

Correct behavior is implemented.

### Possible Drawbacks 

Tests can be refactored.

### Usage Examples or Tests 

postgres_query_executor_test

the case GetAccountTransactionsExecutorTest.ValidPaginationOrder

### CODE REVIEW

File: test/module/irohad/ametsuchi/postgres_query_executor_test.cpp [hunk 1]
```diff
@@ -1119,11 +1120,30 @@ namespace iroha {
...
+        }
+      }
+
       const std::string asset_id = "coin#domain";
       shared_model::crypto::PublicKey fake_pubkey{zero_string};
       shared_model::crypto::Hash hash1;
       shared_model::crypto::Hash hash2;
       shared_model::crypto::Hash hash3;
+      shared_model::crypto::Hash second_block_hash;
+      std::vector<shared_model::crypto::Hash> hashes;
```
Comment (MBoldyrev) [inline test/module/irohad/ametsuchi/postgres_query_executor_test.cpp]: this variable looks weird together with the other ones above (they are all about some hashes!). would it be prettier to return the tx hashes in "additional blocks" form the function that creates these blocks?

File: test/module/irohad/ametsuchi/postgres_query_executor_test.cpp [hunk 2]
```diff
@@ -1340,6 +1360,37 @@ namespace iroha {
...
+
+      auto query =
+          TestQueryBuilder()
+              .creatorAccountId(account_id)
+              .getAccountTransactions(account_id, kTxPageSize, hashes.back())
+              .build();
+      auto result = executeQuery(query);
+      checkSuccessfulResult<shared_model::interface::TransactionsPageResponse>(
+          std::move(result), [](const auto &cast_resp) {
+            ASSERT_EQ(cast_resp.transactions().size(), 1);
```
Comment (MBoldyrev) [inline test/module/irohad/ametsuchi/postgres_query_executor_test.cpp]: why assert and not expect? your code seems apt to continue even when this cheek fails.

File: test/module/irohad/ametsuchi/postgres_query_executor_test.cpp [hunk 3]
```diff
@@ -1340,6 +1360,37 @@ namespace iroha {
...
+            for (const auto &tx : cast_resp.transactions()) {
+              static size_t i = 0;
+              EXPECT_EQ(account_id, tx.creatorAccountId())
+                  << tx.toString() << " ~~ " << i;
+              ++i;
+            }
```
Comment (MBoldyrev) [inline test/module/irohad/ametsuchi/postgres_query_executor_test.cpp]: should we also check the hash of tx?

Comment (igor-egorov) [reply test/module/irohad/ametsuchi/postgres_query_executor_test.cpp]: Fixed.

File: test/module/irohad/ametsuchi/ametsuchi_fixture.hpp [hunk 4]
```diff
@@ -181,19 +181,19 @@ CREATE INDEX IF NOT EXISTS tx_status_by_hash_hash_index ON tx_status_by_hash USI
...
 );
 CREATE TABLE IF NOT EXISTS index_by_creator_height (
     id serial,
     creator_id text,
-    height text,
-    index text
+    height bigint,
+    index bigint
 );
 CREATE TABLE IF NOT EXISTS index_by_id_height_asset (
```
Comment (lebdron) [inline test/module/irohad/ametsuchi/ametsuchi_fixture.hpp]: Can you check why is `index_by_id_height_asset` table is missing in `StorageImpl`, and `position_by_account_asset` is missing here?
https://github.com/hyperledger/iroha/blob/375512635e5619c0346c01ea61c4c902ce1c2639/irohad/ametsuchi/impl/storage_impl.cpp#L990
I think this discrepancy has to be resolved before merging this pull request.

Comment (igor-egorov) [reply test/module/irohad/ametsuchi/ametsuchi_fixture.hpp]: Left a todo

**Pull request merged**